src/plugins/dashboard/public/application/lib/sync_dashboard_index_patterns.ts

```diff
@@ -72,32 +77,8 @@ export const syncDashboardIndexPatterns = ({
     })
   );
 
-  return merge(
-    // output of dashboard container itself
-    dashboardContainer.getOutput$(),
-    // plus output of dashboard container children,
-    // children may change, so make sure we subscribe/unsubscribe with switchMap
-    dashboardContainer.getOutput$().pipe(
```

Comment:
I am not sure I understand why this part where we subscribed to each child's output is no longer needed. 
Could you elaborate, please?

Reply:
Dashboard Container output changes any time the Loading State of any of its child embeddables changes. This is stored in `dashboardContainer.getOutput().embeddableLoaded`. This key used to only be updated **immediately** after a new embeddable was created via its factory. 

Some embeddables like maps & lens don't know their output, or their index patterns immediately after being created. I will call these Deferred Initialization Embeddables. Even in the case of a Deferred Initialization Embeddable, that `embeddableLoading` state was getting populated to `true` right away, which would cause an index pattern poll when the embeddable erroneously didn't have any index patterns to report. This meant that the only way for dashboard to get the true index patterns was to subscribe to the output of each individual embeddable.

Instead, this PR makes these Deferred Initialization Embeddables in charge of communicating to the container when they have finished loading. That means that by the time the dashboard's `embeddableLoaded` output changes, the child embeddable will have finished initialization, and will be able to provide the correct index pattern.

This is the only way I could think of to fix https://github.com/elastic/kibana/issues/105182, and the simplification of the index pattern fetching is a byproduct of that.